scripts/prepare_doc_deploy.ts

```diff
@@ -30,13 +30,18 @@ const folderName = branch !== "master" ? commitHash : "master";
 // │   ├── index.html (and assets for minisite)
 // │   ├── releases.json (list all releases in order)
 
-ensureDirSync("dist/gh_deploy");
-copySync("www/dist/", "dist/gh_deploy/");
+const targetFodlerGh = `dist/gh_deploy`;
+const targetFodlerS3 = `dist/s3_deploy/${folderName}`;
 
-ensureDirSync(`dist/s3_deploy/${folderName}`);
-moveSync("dist/doc", `dist/s3_deploy/${folderName}/doc`);
-moveSync("dist/doc-snippets", `dist/s3_deploy/${folderName}/doc-snippets`);
-moveSync("dist/examples", `dist/s3_deploy/${folderName}/examples`);
+removeSync(targetFodlerGh);
+ensureDirSync(targetFodlerGh);
+copySync("www/dist/", `${targetFodlerGh}/`);
+
+removeSync(targetFodlerS3);
```

Comment:
I think moving the whole `dist/` is risky.
Why don't we do like before? Explicit move of files/folders.
```
mkdir -p dist/gh_deploy
mv dist/index.html dist/gh_deploy
mv dist/css dist/gh_deploy
mv dist/resources dist/gh_deploy
mv dist/js dist/gh_deploy
mv dist/redirect_examples dist/gh_deploy/examples
mv dist/redirect_docs dist/gh_deploy/docs

mv dist/_config.yml dist/gh_deploy

mkdir -p dist/s3_deploy/${folder_name}
mv dist/doc* dist/s3_deploy/${folder_name}
mv dist/examples dist/s3_deploy/${folder_name}
```

Reply:
FThe folder structure has been changed, but anyway in my opinion fix this script by:

1) Don't move things at all now. 
2) Clean target folders, so we don't get old stuff
3) ... finally _copy_ stuff to respective dist folders. 

It's much easier to test and debug such script because it doesn't destroy build artifcats in their original place.
Ignoring these changes, new script does exactly what is required for S3 and GH dist ... let's see that CI says.

Reply:
That is a good point, copying instead of moving. You have my vote for that :)

But, copying the whole `dist` folder is a bit of an overkill as in `gh-pages` we only need the documentation and webpage. So, adding docs and examples is perhaps too much? Well, probably is not that bad, but just bloats a bit the branch. I see that for s3 we only move doc, doc-snippets and examples (as it should be).
Don't want to be too obnoxious about it, just trying to avoid having a too generic rule :)

Reply:
> So, adding docs and examples is perhaps too much?

But examples and docs don't land in `dist/gh-pages`.

The structure is same as done with original script created by you:

* `dist/gh_deploy` -> only "website" and this goes to GH-pages
* `dist/s3_deploy/$ref` -> examples and docs and snippets and this goes to S3

This is how it looks: https://gist.github.com/zbigg/e3b0be3fec1d56f712408011514fbd25

Reply:
Because I need to learn to read. I see you move www/dist to gh_deploy and not dist/ :)
I was stubbornly reading dist and I could not make sense of it.
My apologies, will resolve the conversation.